### PR TITLE
Add filters for escaping variables in YAML

### PR DESCRIPTION
**Pull request by andrew-moore-octo**
This PR adds two new filters for escaping variables in yaml:

- `YamlSingleQuoteEscape`
- `YamlDoubleQuoteEscape`

How to review this PR:

- Code quality
- Do we have enough test cases? Are there any edge cases to consider?
- Do we need any other filters?

### CODE REVIEW

**Comment by andrew-moore-octo** [comment]
[YamlEscape OctoStache filter](https://trello.com/c/ycY0mkor/6-yamlescape-octostache-filter)

File: source/Octostache.Tests/FiltersFixture.cs [hunk 1]
```diff
@@ -90,6 +92,109 @@ public void JsonIsEscaped(string input, string expectedResult, string testName)
...
+        public void YamlSingleQuotedStringsCanRoundTripWithSideEffects(string input, string expected)
+        {
+            var yaml = Evaluate("Key: '#{Input | YamlSingleQuoteEscape}'", new Dictionary<string, string> { { "Input", input } });
+
+            var doc = new DeserializerBuilder()
+                .Build()
+                .Deserialize<TestDocument>(yaml);
+
+            // Yamldotnet normalises \r\n in single quoted scalars to \n.
+            var normalisedExpected = expected.Replace("\r\n", "\n");
```
Comment (andrew-moore-octo) [inline source/Octostache.Tests/FiltersFixture.cs:194]: This is the best approach I could find. I've left the `expected` strings as they are (because they represent what we expect), and then performed a replace on them down here because if I'm interpreting it right, the normalisation of the newlines is a decision made by Yamldotnet, and not a strict requirement of the yaml spec.
